nav2_route/src/plugins/graph_file_loaders/geojson_graph_file_loader.cpp

```diff
@@ -0,0 +1,242 @@
+    Operations operations = convertOperationsFromJson(properties);
+    Metadata metadata = convertMetaDataFromJson(properties);
+
+    graph[graph_to_id_map[start_id]].addEdge(
+      edge_cost, &graph[graph_to_id_map[end_id]], id,
+      metadata, operations);
+  }
+}
+
+Coordinates GeoJsonGraphFileLoader::convertCoordinatesFromJson(const Json & node)
+{
+  Coordinates coords;
+  const auto & properties = node["properties"];
+  if (properties.contains("frame")) {
+    coords.frame_id = properties["frame"];
+  }
+
+  const auto & coordinates = node["geometry"]["coordinates"];
+  coords.x = coordinates[0];
+  coords.y = coordinates[1];
+
+  return coords;
+}
+
+Metadata GeoJsonGraphFileLoader::convertMetaDataFromJson(
+  const Json & properties,
+  const std::string & key)
+{
+  Metadata metadata;
+  if (!properties.contains(key)) {return metadata;}
+
+  for (const auto & data : properties[key].items()) {
+    if (data.value().is_object() ) {
+      Metadata new_metadata = convertMetaDataFromJson(properties[key], data.key());
+      metadata.setValue(data.key(), new_metadata);
+      continue;
+    }
+
+    const auto setPrimitiveType = [&](const auto & value) -> std::any
+      {
+        if (value.is_number()) {
+          if (value.is_number_unsigned()) {
+            return static_cast<unsigned int>(value);
+          } else if (value.is_number_integer()) {
+            return static_cast<int>(value);
+          } else {
+            return static_cast<float>(value);
+          }
+        }
+
+        if (value.is_boolean()) {
+          return static_cast<bool>(value);
+        }
+
+        if (value.is_string()) {
+          return static_cast<std::string>(value);
+        }
+        RCLCPP_ERROR(
+          logger_, "Failed to convert the key: %s to a value", data.key().c_str());
+        throw std::runtime_error("Failed to convert");
+      };
+
+    if (data.value().is_array()) {
+      std::vector<std::any> array;
+      for (const auto & el : data.value()) {
+        auto value = setPrimitiveType(el);
+        array.push_back(value);
+      }
+      metadata.setValue(data.key(), array);
+      continue;
+    }
+
+    auto value = setPrimitiveType(data.value());
+    metadata.setValue(data.key(), value);
+  }
+
+  return metadata;
+}
+
+Operation GeoJsonGraphFileLoader::convertOperationFromJson(const Json & json_operation)
+{
+  Operation operation;
+  json_operation.at("type").get_to(operation.type);
+  Json trigger = json_operation.at("trigger");
+  operation.trigger = trigger.get<OperationTrigger>();
+  Metadata metadata = convertMetaDataFromJson(json_operation);
+  operation.metadata = metadata;
+
+  return operation;
+}
+
+Operations GeoJsonGraphFileLoader::convertOperationsFromJson(const Json & properties)
+{
+  Operations operations;
+  if (properties.contains("operations")) {
+    for (const auto & json_operation : properties["operations"]) {
+      std::cout << "getting operations" << std::endl;
+      Operation operation;
```

Comment:
```suggestion
      operations.push_back(convertOperationFromJson(json_operation));
```

All in 1 line, no need for 3 lines